Install `tbdocs` for the `@web5/api` package

leordev opened the pull request:
- Installs `tbdocs` on `@web5/api`, which will start to bring up doc errors on PRs (powered by api-extractor), [see below](https://github.com/TBD54566975/web5-js/pull/251#issuecomment-1787340652)
- These errors are non blocking, you can feel free to ignore them if you want to, although not encouraged
- Since we aligned on this, I'm helping to fix the initial warnings pointed by tbdocs by fixing and adding missing docs (just a first pass, but please review my docs to see if it makes sense!)
  - I tagged everything as `@beta` but can you please point me to apis that are more stable that we should tag as `@public`, so that we don't break our customers? ([Release tags reference](https://api-extractor.com/pages/tsdoc/doc_comment_syntax/#release-tags))

Fix #250

leordev commented:
> **TBDocs Report**
> 
> 🛑 Errors: 0 ⚠️ Warnings: 23

The above warnings will be fixed at the tbdocs level repo (hence could be ignored)

- The `abstract-level` compilations errors should be suppressed
- The missing documentation attributes are all from private properties that will be fixed with [this api-extractor PR](https://github.com/microsoft/rushstack/pull/4392) (I just need to test their changes and make sure it worked!)

frankhinek commented on packages/api/src/record.ts:
```diff
@@ -107,7 +170,11 @@ export class Record implements RecordModel {
   /**
-   * TODO: Document method.
+   * Returns the data of the current record.
+   * If the record data is null, it attempts to fetch the data from the DWN.
```
```suggestion
   * If the record data is not available, it attempts to fetch the data from the DWN.
```

frankhinek commented on packages/api/src/did-api.ts:
```diff
@@ -28,6 +28,13 @@ import type { Web5Agent } from '@web5/agent';
 // }
+
+/**
+ * The DID API is used to create and resolve DIDs. It is the primary interface
+ * for interacting with DIDs.
```
```suggestion
 * The DID API is used to create and resolve DIDs.
```

frankhinek commented on packages/api/src/dwn-api.ts:
```diff
@@ -171,19 +253,19 @@ export class DwnApi {
 
   /**
-   * TODO: Document namespace.
+   * Records manager, used to read, write, query and delete records
```
```suggestion
   * API to to interact with DWN records (e.g., `dwn.records.create()`).
```

frankhinek commented on packages/api/src/dwn-api.ts:
```diff
@@ -109,12 +191,12 @@ export class DwnApi {
   /**
- * TODO: Document namespace.
- */
+   * Retrieves the protocols manager, used to configure and query them
```
```suggestion
   * API to to interact with DWN protocols (e.g., `dwn.protocols.configure()`).
```

frankhinek reviewed (changes requested):
@leordev Thanks for getting this work started and contributing all of the much needed comments! 😄

frankhinek commented:
@leordev Can you also rebase on main to clear the `security-audit` check failure?   The `package-lock.json` in `main` already has the patched version of `browserify-sign` updated.  Thanks.

leordev commented:
Thank you @frankhinek ! Just rebased and implemented the suggestions. 

Did you check if all the interfaces that I tagged as `@beta` but should already be in a stable enough state to be considered `@public`? 

If all of them should be `@beta`, feel free to go ahead and merge!

frankhinek commented on packages/api/src/index.ts:
```diff
@@ -1,8 +1,33 @@
\ No newline at end of file
+export * from './tech-preview.js';
+
+import * as utils from './utils.js';
+export { utils };
```
@leordev Is this limitation with `api-extractor` something that is planned to be resolved?  Seems unfortunate that we have to modify the source code to satisfy `api-extractor` being unable to understand valid ES2020 syntax.

frankhinek commented:
> Thank you @frankhinek ! Just rebased and implemented the suggestions.

Thanks! 👍 
 
> Did you check if all the interfaces that I tagged as `@beta` but should already be in a stable enough state to be considered `@public`?
> 
> If all of them should be `@beta`, feel free to go ahead and merge!

I think `@beta` is fine for now.  We've got some changes upcoming to re-align the Web5 SDKs implemented in TS, Kotlin, and Swift/Rust so that might introduce changes.

frankhinek reviewed (approved):
LGTM! 🚀 Let's ship it!
